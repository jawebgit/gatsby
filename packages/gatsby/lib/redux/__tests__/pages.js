const { store, reducer } = require("../index");
const { actions, boundActions } = require("../actions");

describe(`Add pages`, () => {
  it(`allows you to add pages`, () => {
    const action = actions.upsertPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
    });
    const state = reducer({ pages: [] }, action);
    expect(action).toMatchSnapshot();
    expect(state).toMatchSnapshot();
  });

  it(`allows you to add pages with context`, () => {
    const action = actions.upsertPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
      context: {
        id: 123,
      },
    });
    const state = reducer({ pages: [] }, action);
    expect(action).toMatchSnapshot();
    expect(state).toMatchSnapshot();
  });

  it(`allows you to add multiple pages`, () => {
    const action = actions.upsertPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
    });
    const action2 = actions.upsertPage({
      path: `/hi/pizza/`,
      component: `/whatever/index.js`,
    });
    let state = reducer({ pages: [] }, action);
    state = reducer(state, action2);
    expect(state).toMatchSnapshot();
    expect(state.pages.length).toEqual(2);
  });

  it(`allows you to update existing pages (based on path)`, () => {
    const action = actions.upsertPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
    });

    // Change the component
    const action2 = actions.upsertPage({
      path: `/hi/`,
      component: `/whatever2/index.js`,
    });

    let state = reducer({ pages: [] }, action);
    state = reducer(state, action2);
    expect(state).toMatchSnapshot();
    expect(state.pages.length).toEqual(1);
  });
});
