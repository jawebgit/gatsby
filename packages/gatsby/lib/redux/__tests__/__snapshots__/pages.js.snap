exports[`Add pages allows you to add multiple pages 1`] = `
Object {
  "pages": Array [
    Object {
      "component": "/whatever/index.js",
      "componentChunkName": "page-component---whatever-index-js",
      "path": "/hi/",
    },
    Object {
      "component": "/whatever/index.js",
      "componentChunkName": "page-component---whatever-index-js",
      "path": "/hi/pizza/",
    },
  ],
}
`;

exports[`Add pages allows you to add pages 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "page-component---whatever-index-js",
    "path": "/hi/",
  },
  "type": "UPSERT_PAGE",
}
`;

exports[`Add pages allows you to add pages 2`] = `
Object {
  "pages": Array [
    Object {
      "component": "/whatever/index.js",
      "componentChunkName": "page-component---whatever-index-js",
      "path": "/hi/",
    },
  ],
}
`;

exports[`Add pages allows you to add pages with context 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "page-component---whatever-index-js",
    "context": Object {
      "id": 123,
    },
    "path": "/hi/",
  },
  "type": "UPSERT_PAGE",
}
`;

exports[`Add pages allows you to add pages with context 2`] = `
Object {
  "pages": Array [
    Object {
      "component": "/whatever/index.js",
      "componentChunkName": "page-component---whatever-index-js",
      "context": Object {
        "id": 123,
      },
      "path": "/hi/",
    },
  ],
}
`;

exports[`Add pages allows you to update existing pages (based on path) 1`] = `
Object {
  "pages": Array [
    Object {
      "component": "/whatever2/index.js",
      "componentChunkName": "page-component---whatever-2-index-js",
      "path": "/hi/",
    },
  ],
}
`;
